### PR TITLE
Update cmake requirement

### PR DESCRIPTION
**Pull request by MicK7**
- Update cmake requirement to be coherent with hdf5.
  This could also help with packaging on the long run.

- Fix Fortran to C mapping on Apple ARM64
 This is compiler dependent so a better design based on ISO C should be possible.
 The current correction is based on https://gcc.gnu.org/onlinedocs/gcc-8.1.0/gfortran.pdf  (see hidden parameter section)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
   build:
     strategy:
       matrix:
-        name: ["Ubuntu Latest GCC", "macOS Latest Clang"]
+        name: ["Ubuntu Latest GCC", "macOS Latest"]
```
Comment (brtnfld) [inline .github/workflows/main.yml:23]: Maybe you should use macos-13, which is x86, so we can still test Fortran on Macs.

Comment (MicK7) [reply .github/workflows/main.yml:23]: Indeed a separate runner is needed for x86 Mac. But my goal is to achieve M1/M2/M3 compatibility with Fortran enabled. I have deactivated temporary Fortran and will try to solve the issue as soon as possible. So far, I suspect that transfer of strings between C and Fortran on arm64 may be part of the problem... and I have not figured out why.
I will try to enable an interactive github action to investigate if possible.

Comment (brtnfld) [reply .github/workflows/main.yml:23]: Ok, thanks. I did not realize you were still looking into it. I think flang is now being used, which could be part of the issue. flang was at least an issue for HDF5 Fortran on macs in general.

Comment (MicK7) [reply .github/workflows/main.yml]: @brtnfld Problem solved \o/

The solution works with clang and gcc.

**Comment by MicK7** [comment]
@brtnfld Please review

The branch use clang bug gcc has also been tested (I kept commented the github action line to do it)
Flang is not tested and will require a bit more investigation.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.20)
```
Comment (brtnfld) [inline CMakeLists.txt:1]: Is there a CMake feature that requires the version bump?

Comment (MicK7) [reply CMakeLists.txt:1]: Yes, it is to be coherent with recent HDF5 and also from this cmake version the linking process to HDF5 (finding dependencies, packaging to transfer to consumer code the dependency) has been modified to be more reliable (feature https://cmake.org/cmake/help/v3.20/module/FindHDF5.html).

File: appveyor.yml [hunk 3]
```diff
@@ -33,9 +33,9 @@ skip_tags: true
 #---------------------------------#
 environment:
   global:
-    1_8: &1_8 1.8.21
-    1_10: &1_10 1.10.6
-    1_12: &1_12 1.12.0
+    1_8: &1_8 1.8.23
+    1_10: &1_10 1.10.10
+    1_12: &1_12 1.12.3
```
[the remark this answers is not in the document]
Comment (MicK7) [reply appveyor.yml]: Thanks for the review. I replaced 1.12 by 1.14. Now waiting for the CI to complete to validate this change.

**Review by brtnfld — APPROVED** [review]
LGTM, thanks!

**Pull request merged**